src/huggingface_hub/hub_mixin.py

```diff
@@ -289,37 +289,48 @@ def push_to_hub(
 
 class PyTorchModelHubMixin(ModelHubMixin):
     """
-    Implementation of [`ModelHubMixin`] to provide model Hub upload/download capabilities to PyTorch models. The model
-    is set in evaluation mode by default using `model.eval()` (dropout modules are deactivated). To train the model,
-    you should first set it back in training mode with `model.train()`.
+    This class allows to add `from_pretrained` and `push_to_hub` capabilities to any custom PyTorch model.
+    One simply needs to inherit from this class in order to add these methods to any `nn.Module`.
 
-    Example:
+    It is advised to create an `nn.Module` which takes in a `config` argument in its `__init__` method, which
+    is a Python dictionary containing all hyperparameters such as the number of channels, hidden size, etc.
+
+    When calling `save_pretrained` or `push_to_hub`, pass along the config dictionary. This will then be saved
+    alongside the model weights in a `config.json` file, similar to how this is done in the Transformers library.
+
+    Usage is as follows:
 
     ```python
     >>> import torch
     >>> import torch.nn as nn
     >>> from huggingface_hub import PyTorchModelHubMixin
 
-
     >>> class MyModel(nn.Module, PyTorchModelHubMixin):
-    ...     def __init__(self):
+    ...     def __init__(self, config):
     ...         super().__init__()
-    ...         self.param = nn.Parameter(torch.rand(3, 4))
-    ...         self.linear = nn.Linear(4, 5)
+    ...         self.param = nn.Parameter(torch.rand(num_channels, hidden_size))
+    ...         self.linear = nn.Linear(hidden_size, num_classes)
```

Comment:
cc @Wauplin it would be great if we could do things like `config.num_channels`, `config.hidden_size` (i.e. access all config keys as attributes). For that, I assume, we would need to make sure models do super()__init__(config)